sr: use site replicator svcacct to sign STS session tokens (#19111)
Minio-io/minio: docs/site-replication/README.md

```sh
mc admin replicate info minio1
```

** Note **

> Previously, site replication required root credentials of peer sites be identical. This is no longer required due to STS tokens now being signed with the site replicator service account credentials. However, this means that STS tokens signed previously by root credentials will no longer be valid upon upgrade to the latest version with this change. Also, if site replication is ever removed - the STS tokens previously signed by site replicator service account credentials will no longer be valid.

Reviewer: ```suggestion
Previously, site replication required root credentials of peer sites be identical. This is no longer required due to STS tokens now being signed with the site replicator service account credentials. However, this means that STS tokens signed previously by root credentials will no longer be valid upon upgrade to the latest version with this change. Also, if site replication is ever removed - the STS tokens previously signed by site replicator service account credentials will no longer be valid.
```

We should re-word it to indicate why it is done in this manner, basically to allow the flexibility of having different "root" accounts per site. As well as eventually disabling this root account for all practical purposes.

```sh
mc admin replicate info minio1
```

** Note **

> Previously, site replication required root credentials of peer sites be identical. This is no longer required due to STS tokens now being signed with the site replicator service account credentials vs the root credentials, thus allowing flexibility in independent manageement of root accounts across sites, as well as the ability to disable root account eventually.

Reviewer: ```suggestion
Previously, site replication required the root credentials of peer sites to be identical. This is no longer necessary because STS tokens are now signed with the site replicator service account credentials, thus allowing flexibility in the independent management of root accounts across sites and the ability to disable root accounts eventually.
```

>  However, this means that STS tokens signed previously by root credentials will no longer be valid upon upgrade to the latest version with this change. Also, if site replication is ever removed - the STS tokens previously signed by site replicator service account credentials will no longer be valid.

Reviewer: ```suggestion
However, this means that STS tokens signed previously by root credentials will no longer be valid upon upgrading to the latest version with this change. Please re-generate them as you usually do. Additionally, if site replication is ever removed - the STS tokens will become invalid, regenerate them as you usually do.
```
